sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -530,7 +530,8 @@ export function getGeneratedClientOptions(
       syncTokenPolicy(syncTokens),
       ...retryPolicies,
       ...defaults,
-    ],
+    ],    
+    generateClientRequestIdHeader: true,
```

Comment:
Sorry if I missed this, but where does this actually get passed to? Is there already a policy that reads the option?

Also since this is hard-coded at true it looks like you can't provide a custom client request id, but you have a test for that below. Is this something we need to support?

Reply:
You didn't miss anything, the actual code that reads this setting is in the core-http `ServiceClient`.

https://github.com/Azure/azure-sdk-for-js/blob/418fd86c89d8353b1678d98ff6ce52d37ecb660c/sdk/core/core-http/lib/serviceClient.ts#L592

The client request ID policy is smart enough to only add the header if it doesn't already exist:

https://github.com/Azure/azure-sdk-for-js/blob/418fd86c89d8353b1678d98ff6ce52d37ecb660c/sdk/core/core-http/lib/policies/generateClientRequestIdPolicy.ts#L34

So we're covered in the case that the user does want to pass their own client request ID. 

The nock tests cover both these cases.

Reply:
> The client request ID policy is smart enough to only add the header if it doesn't already exist:

I think I'm missing something because your 2nd test seems to set the request id:
https://github.com/Azure/azure-sdk-for-js/pull/6568/files#diff-bd889982f342edc184a4392a0f514b14R147

Yet you're verifying it matches a string of a different format:
https://github.com/Azure/azure-sdk-for-js/pull/6568/files#diff-bd889982f342edc184a4392a0f514b14R140

Reply:
Is it possible the nock scope from the first test is being used when making the request from your 2nd test?

Reply:
You missed NOTHING and saw EVERYTHING!

I fixed it. The issue was the regex, when not anchored, just does a "anywhere" match. So even though I didn't include ' ' in my list it was able to find an alpha-numeric match elesewhere in the string so we were good.

Fixed by:
1. Anchoring the regex
2. Removing the custom header (test immediately started failing).